e2e/app/page/notebook-preview-page.ts

```diff
@@ -19,20 +19,21 @@ export default class NotebookPreviewPage extends AuthenticatedPage {
   }
 
   async isLoaded(): Promise<boolean> {
-    await waitWhileLoading(this.page);
     await Promise.all([
       this.page.waitForXPath(Selector.editLink, { visible: true }),
       this.page.waitForXPath(Selector.runPlaygroundModeLink, { visible: true }),
       this.page.waitForXPath(Selector.previewLink, { visible: true })
     ]);
+    await this.waitForNotebookCellsRendered();
```

Comment:
Ensure Preview page is ready: Wait until all `code_cell` are rendered in notebook iframe.